### PR TITLE
Migration from nereid-catalog to channel listing for webshop

### PR DESCRIPTION
**Pull request by riteshshrv**

### CODE REVIEW

File: product.py [hunk 1]
```diff
@@ -72,6 +77,60 @@ def __setup__(cls):
...
+        # products = cls.search([
+        #     ('channel_listings.channel.source', '=', 'webshop'),
+        #     ('channel_listings.channel', '=', current_channel),
+        # ])
+
+        return [('id', 'in', map(int, products))]
+
+    def get_uri(self, name):
+        return self.channel_listings[0].uri
+
```
Comment (priyankarani) [inline product.py]: Channel listing may not always be there

File: product.py [hunk 2]
```diff
@@ -72,6 +77,57 @@ def __setup__(cls):
...
+    def search_displayed_on_eshop(cls, name, domain):
+        current_channel = Transaction().context.get('current_channel')
+        products = cls.search([
+            ('channel_listings.channel.source', '=', 'webshop'),
+            ['OR', [
+                ('channel_listings.channel', '=', current_channel)
+            ],
+                [
+                    ('channel_listings.channel', '=',
+                        request.nereid_website.channel)
```
Comment (prakashpp) [inline product.py]: this code will not work outside the request context, remove this clause. Current channel and source checking clause is enough

File: product.py [hunk 3]
```diff
@@ -72,6 +77,57 @@ def __setup__(cls):
...
+            ]
+        ])
+
+        return [('id', 'in', map(int, products))]
+
+    def get_uri(self, name):
+        return self.channel_listings and self.channel_listings[0].uri or None
+
+    @classmethod
+    def search_uri(cls, name, clause):
```
Comment (prakashpp) [inline product.py]: not needed

File: product.py [hunk 4]
```diff
@@ -72,6 +77,57 @@ def __setup__(cls):
...
+
+    def _get_webshop_listing(self):
+        """
+        Returns webshop listing for current product
+        """
+        ProductListing = Pool().get('product.product.channel_listing')
+
+        return ProductListing.search([
+            ('product', '=', self.id),
+            ('channel.source', '=', 'webshop')
```
Comment (prakashpp) [inline product.py]: check for current channel

**Pull request closed without merging**